From: Wei Yongjun <weiyj.lk@gmail.com>
Date: Mon, 15 Aug 2016 22:51:04 +0000
Subject: net: ena: Fix error return code in ena_device_init()
Patch-mainline: v4.9-rc1
Git-commit: 6e22066fd02b675260b980b3e42b7d616a9839c5
References: bsc#1027968

Fix to return a negative error code from the invalid dma width
error handling case instead of 0.

Signed-off-by: Wei Yongjun <weiyj.lk@gmail.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Benjamin Poirier <bpoirier@suse.com>
---
 drivers/net/ethernet/amazon/ena/ena_netdev.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/net/ethernet/amazon/ena/ena_netdev.c
+++ b/drivers/net/ethernet/amazon/ena/ena_netdev.c
@@ -2329,6 +2329,7 @@ static int ena_device_init(struct ena_co
 	dma_width = ena_com_get_dma_width(ena_dev);
 	if (dma_width < 0) {
 		dev_err(dev, "Invalid dma width value %d", dma_width);
+		rc = dma_width;
 		goto err_mmio_read_less;
 	}
 
